app/src/main/java/fr/free/nrw/commons/nearby/fragments/NearbyParentFragment.java

```diff
@@ -260,6 +270,42 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
                 scaleBarPlugin.create(scaleBarOptions);
             });
         });
+
+        ClickableSpan spanMapBox=new ClickableSpan() {
+            @Override
+            public void onClick(@NonNull View widget) {
+                Intent intent = new Intent(Intent.ACTION_VIEW);
+                intent.setData(Uri.parse("https://www.mapbox.com/about/maps/"));
+                startActivity(intent);
+            }
+        };
+
+        ClickableSpan spanOpenStreet=new ClickableSpan() {
+            @Override
+            public void onClick(@NonNull View widget) {
+                Intent intent = new Intent(Intent.ACTION_VIEW);
+                intent.setData(Uri.parse("http://www.openstreetmap.org/copyright"));
+                startActivity(intent);
+            }
+        };
+
+        ClickableSpan spanImproveThisMap=new ClickableSpan() {
+            @Override
+            public void onClick(@NonNull View widget) {
+                Intent intent = new Intent(Intent.ACTION_VIEW);
+                intent.setData(Uri.parse("https://www.mapbox.com/map-feedback/"));
+                startActivity(intent);
+            }
+        };
+
+        SpannableString spannableString = new SpannableString(tvAttribution.getText());
+        spannableString.setSpan(spanMapBox, 0,6, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+        spannableString.setSpan(spanOpenStreet, 11,24, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+        spannableString.setSpan(spanImproveThisMap, 27,43, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+        spannableString.setSpan(new ForegroundColorSpan(Color.WHITE),0,tvAttribution.getText().length(),Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
+        tvAttribution.setText(spannableString);
+        tvAttribution.setMovementMethod(LinkMovementMethod.getInstance());
+
```

Comment:
With the CDATA string that has in-built links, the following is the code that was needed here:

 ```suggestion

         tvAttribution.setText(StringUtil.fromHtml(getResources().getString(R.string.map_attribution)));
         tvAttribution.setMovementMethod(LinkMovementMethod.getInstance());
```